config/swarm-sip.yml

```diff
@@ -49,55 +46,14 @@ kdump_enable: false
+# The user is fedora for Kubernetes and Docker Swarm container clusters.
+alaska_admin_user: fedora
 
-# SIP sudoers
-sudoers_sudoers:
-  privileges:
-  - name: "%wheel"
-    entry: "ALL=NOPASSWD:ALL"
+# Extra administrators for the cluster
+extra_admins:
+  - username: "{{ alaska_admin_user }}"
+    ssh_key:
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQCeGbpm1SHEdEEquMw+xyrZ0TqULQ1aL4LuDsbF3NQedAxyg+1/PvY3/+HwSs8dZUlP/DGVrgPdP7JEfKi14oF4TQ6oIy89bKkH0QUF5DkIP8g6cHTZZjZxw1Gso47N636u1qw46dSKiBrYFMmEzRMbPVOWGzxdYHP6FSryrl6YiDtiegBA6/j6lM3a0DdUHtV10KhU58NyNl4MSVyQrayqYtlqZK05QFQ47lpE8FHU0Ud6AHB4W4D8Hy0lbUeIVN+S9xZWKRvCmyQSeIX9mLL3rAvDgFYLbGxYCQ4GTnG1U/Ky2CpZBh1n5ihtDINuXDN0uk9YWNMtAGjm4YUqhSopgLKC3EY/fXtZ28Nr3AMtggBD+vY+FnemSFegoTzKwExmTDBEAzTjZK40parklOBQYimgy4iBeD7//iVis/RFKw1vAKLUCPKkx/eSkLhr9e1IqzKcPIWF4hYVRTGMdM42ziJZ/DHiGcOWKRxDzdhM/wT8SZ0xN7H5h8g7pn+SAmWQETLX4I2qvY2qXx0RcoXzB/cNvAOqxtXFTYZZi1nhzZ4u4S6hD1zOalxg6r5CT5VgqL5AHoe3EmFcvD5t0loGWO1e7CqHV0KOCj9jzOTju+BlQbZEPo8zcZfWlpE+s807NkRDqr1Obg2Sbc6kPPtGlNJcxlDMQymHTPnxgH4e9Q== hpcmort1@ilab-gate.ilab.cluster"
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDeHHJ2Thsnc/F4bx6ZjjKZV/BX68c3Xnl40Aie69wJ00Uh1maYP1p6Pj1QSreFZbYhx11XqwD1a6rlPASAJe/cYj4PIsvmbgAGZqzEpXsWqJij/R7BaYm7EoWiqx4iI7jXCCDE8QJfZC4gR2SPhD3ClLc07SCuvOm6OABJq+KUih0jrkLPcNNK9xiufhe/gP2lfiy7aUPBZThkd6AVgWV+nHaK2dm7d532ChARSpB/Kr3bsyONFDI0+sCxUDtcEHblRw/VVubMLir6UTsYchtCDVVlR3fuiDVM4Oj/p/P8P9OE5VuJ6le1gkYKVK1JFA9LqgPGsbYsuhvylDUe8Acl Generated-by-Nova"
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQClobEi5Eg5GAZk0o43Vb+KgAh6UmTm+rtmaLXVBTZp+e3BubDGK2RVNzmLgXVsu+9P7YdMC65c9E8iC2WXkLmdjTbXYRqCbgiMs6/MCe40YIhV/iBZvyBS8Rb46VKGPBjfJIvU9P2Cs7CYA6Uh0/Vyg1LfFDfq3SfzNBUK5/+gSzR2DyB6hptazq40qa9z/StZ8Qsa0xvIFEV5+eDqPB51MBhZLn3tjy+4tWNRMIgx2CQllYt38w/dtFsLcZvXHEyfm3fYjqfemWDJa9PNMpGB4lSRKSskH4+sVLDGRpCoGp10EXkcfYwjGkNaFA+ZfSpy0Sc0VEPDpdpg+H7tM9dvPHICrJ7ITQlgElKRsUX9H8H/rFsgqf18dWg9rF5Zi4wikSQR30JZbwHZUjhRamq/rJ6LbmfYjMB6tzr9RTyZVT8V8HO9BVvxO+3erIMpSALc6c9wgTxI5BTdhKDInW1uX2OY73RkUhWtCR7QjrpefyS7EI/ya8RvorXcKUxGj1E60Cjcl5EvbuSZk7Z51CKlK2w5uquDamJnqjUCqGzrnkMaIBlVXzfDXVy4VQnAETiC/oeGmx+qQrc3bqkqcg0xADWd9wWvo1rLA83Wh/KGCNkCUK6Wekpi9B+Cv5xUq9GBhGwQ5Pqhzy4WtcOPhrdWPnhf0ZElVL+XovEQKxKq0Q== hpcdulw1@ilab-gate.ilab.cluster"
```

Comment:
This isn't doing the same thing. Previously non-admins had their own user account, with fixed UIDs for consistent filesystem access. This just adds everyone's SSH keys to the admin user.

Reply:
Okay, we don't use this pattern in cluster-infra-clusters, but use the centos user for all admin access, don't you think its better to carry the same patter over to container-infra?

Reply:
For cluster-infra we have users.yml which adds all users in alaska_users, and cluster-admins.yml which adds SSH keys for cluster_admins to the admin user account.

Reply:
Thats right, this is a rather special case where these users need admin rights now for tuning but will not do in the long term. They are not ordinary "users" either as in the case of cluster-infra as this is not how we anticipate they will interact with the cluster if we're going down the Magnum route. Hence extra_admins seems most appropriate.